frontend/components/notifications/SubscriptionRow.vue

```diff
@@ -0,0 +1,225 @@
+}
+
+function receiveFromVModel (state: CheckboxAndNumber) : CheckboxAndText {
+  const output = {} as CheckboxAndText
+  output.check = state.check
+  if (type.value === 'amount' || type.value === 'percent') {
+    output.text = isNaN(state.num) ? '' : String(state.num)
+  }
+  return output
+}
+
+function sendToVModel (state: CheckboxAndText) : CheckboxAndNumber {
+  const output = {} as CheckboxAndNumber
+  output.check = state.check
+  if (type.value === 'amount' || type.value === 'percent') {
+    const corrected = correctUserInput(state.text)
+    output.num = (corrected === '') ? NaN : Number(corrected)
+  }
+  return output
+}
+
+// input validation / autocorrection
+
+function correctUserInput (input: string) : string {
+  switch (type.value) {
+    case 'amount' : return isThisAvalidInput(input) ? String(calculateCorrectNumber(input)) : ''
+    case 'percent' : return String(calculateCorrectNumber(input))
+  }
+  return ''
+}
+
+function calculateCorrectNumber (input: string) : number {
+  let num = !isThisAvalidInput(input) ? Math.abs(props.default ?? 0) : Number(input)
+  if (type.value === 'percent') {
+    if (num < 1) { num = 1 }
+    if (num > 100) { num = 100 }
+    num = Math.round(10 * num) / 10
+  }
+  return num
+}
+
+function isThisAvalidInput (input: string) : boolean {
+  return !!input && !isNaN(Number(input)) && Number(input) >= 0
+}
+
+// calculation of the information to show in the tooltip
+
+const tooltipLines = computed(() => {
+  let options
+  if (props.valueInText !== undefined) {
+    options = { plural: props.valueInText }
+  } else
+    if (Array.isArray(state.value)) {
+      options = { plural: state.value.length, list: state.value.join(', ') }
+    } else {
+      let plural: number
+      if (type.value === 'amount' || type.value === 'percent') {
+        plural = calculateCorrectNumber(checkboxAndText!.value.text)
+      } else {
+        plural = state.value ? 2 : 1
+      }
+      options = { plural }
+    }
+  return tAll(t, props.tPath + '.hint', options)
+})
+
+// output: on blur or enter, auto-correcting the text field
+function acknowledgeInputtedText () {
+  checkboxAndText!.value.text = correctUserInput(checkboxAndText!.value.text)
+}
+
+const deactivationClass = props.lacksPremiumSubscription ? 'deactivated' : ''
+</script>
+
+<template>
+  <div class="option-row">
+    <span class="caption" :class="deactivationClass">
+      {{ t(tPath+'.option') }}
+    </span>
+    <BcTooltip v-if="tooltipLines[0]" :fit-content="true">
+      <FontAwesomeIcon :icon="faInfoCircle" class="info" />
+      <template #tooltip>
+        <BcMiniParser :input="tooltipLines" class="tt-content" />
+      </template>
+    </BcTooltip>
+    <BcPremiumGem v-if="lacksPremiumSubscription" class="gem" />
+    <div v-if="type != 'networks'" class="right">
+      <div v-if="type == 'amount' || type == 'percent'" class="input">
+        <InputText
```

Comment:
This would not remove the validation logic because it is used by the bridge to make sure that the real-time changes that we send to the API are valid.

For example, when the user types `952` in the `%` input, our specification makes us output something to the API immediately (I mean with debouncing) but the automatic correction of `<InputNumber>` will kick in only if the user unfocuses.
The callbacks of the bridge need the correcting functions.